### PR TITLE
Added Expiration date page UI

### PR DESCRIPTION
**Pull request by surabhishukla**

### CODE REVIEW

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 1]
```diff
@@ -2,9 +2,22 @@ import Grid from "@mui/material/Grid";
...
+import { FormControlLabel } from "@mui/material";
+import Checkbox from "@mui/material/Checkbox";
+import Radio from "@mui/material/Radio";
+import { RadioGroup } from "@mui/material";
+import { useContext, useState } from "react";
+import { AppContext } from "./../../../App";
 
 const LimitationDetails = () => {
+  const { state, setState } = useContext(AppContext);
+  const [radioValue, setRadioValue] = useState("No expiry Date");
```
[the remark this answers is not in the document]
Comment (surabhishukla) [reply src/Pages/License/Generate/LimitationDetails.jsx:14]: done

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 2]
```diff
@@ -2,9 +2,22 @@ import Grid from "@mui/material/Grid";
...
+
+  const radioHandler = (e) => {
+    setRadioValue(e.target.value);
+    console.log(e.target.value);
```
[the remark this answers is not in the document]
Comment (surabhishukla) [reply src/Pages/License/Generate/LimitationDetails.jsx]: done

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 3]
```diff
@@ -17,17 +30,111 @@ const LimitationDetails = () => {
...
+                  <TextField
+                    disabled={radioValue !== "Expiry in days:"}
+                    label="Number of days"
+                    variant="outlined"
+                    size="small"
+                    required
+                    value={state.limitation.expiration}
+                    onChange={(e) =>
+                      setState({
+                        ...state,
```
[the remark this answers is not in the document]
Comment (surabhishukla) [reply src/Pages/License/Generate/LimitationDetails.jsx]: done

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 4]
```diff
@@ -17,17 +30,111 @@ const LimitationDetails = () => {
...
+            <Grid item mr={3}>
+              <TextField
+                label="Limit Count"
+                variant="outlined"
+                size="small"
+                required
+                value={state.limitation.limitCount}
+                onChange={(e) =>
+                  setState({
+                    ...state,
```
[the remark this answers is not in the document]
Comment (surabhishukla) [reply src/Pages/License/Generate/LimitationDetails.jsx]: done

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 5]
```diff
@@ -2,9 +2,22 @@ import Grid from "@mui/material/Grid";
 import Typography from "@mui/material/Typography";
 import TextField from "@mui/material/TextField";
 import { CustomCard, CustomCardContent } from "./GenerateLicense";
-
+import { FormControlLabel } from "@mui/material";
+import Checkbox from "@mui/material/Checkbox";
+import Radio from "@mui/material/Radio";
+import { RadioGroup } from "@mui/material";
+import { useContext, useState } from "react";
+import { AppContext } from "./../../../App";
```
[the remark this answers is not in the document]
Comment (surabhishukla) [reply src/Pages/License/Generate/LimitationDetails.jsx]: done

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 6]
```diff
@@ -2,9 +2,45 @@ import Grid from "@mui/material/Grid";
...
 
 const LimitationDetails = () => {
+  const { state, setState } = useContext(AppContext);
+  const [radioValue, setRadioValue] = useState("No expiry Date");
+
+  const radioHandler = (e) => {
+    setRadioValue(e.target.value);
+  };
+
+  const EXPIRYDAYS = "Expiry in days:";
```
Comment (pinak-globallogic) [inline src/Pages/License/Generate/LimitationDetails.jsx:20]: Put constants outside of component

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 7]
```diff
@@ -2,9 +2,45 @@ import Grid from "@mui/material/Grid";
...
+  const NOEXPIRY = "No expiry Date";
+  const SPECIFICDATE = "Specific date";
+
+  const limitCountValue = (e) => {
```
Comment (pinak-globallogic) [inline src/Pages/License/Generate/LimitationDetails.jsx:24]: onLimitCountChange

File: src/Pages/License/Generate/LimitationDetails.jsx [hunk 8]
```diff
@@ -2,9 +2,45 @@ import Grid from "@mui/material/Grid";
...
+    setState({
+      ...state,
+      limitation: {
+        ...state.limitation,
+        limitCount: e.target.value,
+      },
+    });
+  };
+
+  const expirationValue = (e) => {
```
Comment (pinak-globallogic) [inline src/Pages/License/Generate/LimitationDetails.jsx:34]: onExpirationChange

**Review by pinak-globallogic — CHANGES_REQUESTED** [review]

File: src/Pages/License/Generate/LimitationDetails.jsx (hunk 8 above)
Comment (surabhishukla) [reply src/Pages/License/Generate/LimitationDetails.jsx]: done

File: src/Pages/License/Generate/LimitationDetails.jsx (hunk 7 above)
Comment (surabhishukla) [reply src/Pages/License/Generate/LimitationDetails.jsx]: done

**Review by pinak-globallogic — APPROVED** [review]

**Pull request merged**